elements/pl-order-blocks/pl-order-blocks.py

```diff
@@ -5,175 +5,167 @@
-
-            is_correct = pl.get_boolean_attrib(html_tags, 'correct', PL_ANSWER_CORRECT_DEFAULT)
-            if check_indentation is False:
-                try:
-                    answer_indent = pl.get_string_attrib(html_tags, 'indent')
-                except Exception:
-                    answer_indent = -1
-                else:
-                    raise Exception('<pl-answer> should not specify indentation if indentation is disabled.')
-            else:
-                answer_indent = pl.get_integer_attrib(html_tags, 'indent', PL_ANSWER_INDENT_DEFAULT)  # get answer indent, and default to -1 (indent level ignored)
-            if is_correct is True:
-                # add option to the correct answer array, along with the correct required indent
-                if pl.get_string_attrib(html_tags, 'ranking', '') != '':
-                    ranking = pl.get_string_attrib(html_tags, 'ranking')
-                    try:
-                        ranking = int(ranking) - 1
-                    except ValueError:
-                        raise Exception('Ranking specified in <pl-answer> is not a number.')
-                    correct_answers_ranking.append(ranking)
-                correct_answers.append(html_tags.text)
-                correct_answers_indent.append(answer_indent)
-            else:
-                incorrect_answers.append(html_tags.text)
-            html_ordering.append(html_tags.text)
+    index = 0
+    for html_tags in element:  # iterate through the html tags inside pl-order-blocks, should be all <pl-answer> tags
+        if html_tags.tag != 'pl-answer':
+            raise Exception('Any html tags nested inside <pl-order-blocks> must be <pl-answer>.')
+
+        if grading_method == 'external':
+            pl.check_attribs(html_tags, required_attribs=[], optional_attribs=['correct'])
+        else:
+            pl.check_attribs(html_tags, required_attribs=[], optional_attribs=['correct', 'ranking', 'indent'])
+
+        is_correct = pl.get_boolean_attrib(html_tags, 'correct', PL_ANSWER_CORRECT_DEFAULT)
+        answer_indent = pl.get_integer_attrib(html_tags, 'indent', None)
+        inner_html = pl.inner_html(html_tags)
+        ranking = pl.get_integer_attrib(html_tags, 'ranking', -1) - 1
+
+        answer_data_dict = {'inner_html': inner_html,
+                            'indent': answer_indent,
+                            'ranking': ranking,
+                            'index': index}
+
+        if check_indentation is False and answer_indent is not None:
+            raise Exception('<pl-answer> should not specify indentation if indentation is disabled.')
+
+        if is_correct:
+            correct_answers.append(answer_data_dict)
         else:
-            raise Exception('Tags nested inside <pl-order-blocks> must be <pl-answer>.')
+            incorrect_answers.append(answer_data_dict)
+        index += 1
```

Comment:
I always love to see code get shorter and simpler :)

Reply:
:sunglasses: